### PR TITLE
INFRA-1801: Fix serialization of Error objects

### PR DESCRIPTION
**Pull request by burnsed**
Fix for INFRA-1801

### CODE REVIEW

File: lib/kayvee.ts [hunk 1]
```diff
@@ -3,12 +3,27 @@ _.mixin(require("underscore.deep"));
 
 const deploy_env = process.env._DEPLOY_ENV;
 
+// Encode errors using their properties instead of toJSON()
+function replaceErrors(key, value) {
+  if (value instanceof Error) {
+    var error = {};
+
+    _.each(Object.getOwnPropertyNames(value), (k) => {
+      error[k] = value[k];
```
Comment (mohit) [inline lib/kayvee.ts]: Do we really want all the parameters? Sorry but I'm not sure what it usually looks like

**Comment by mohit** [comment]
Just curious about use of err.toString() as opposed to the whole dict

**Comment by burnsed** [comment]
@mohit : thats a good point. For the purpose of kayvee logging, toString() is probably more appropriate. I'll update

File: lib/kayvee.ts [hunk 2]
```diff
@@ -3,12 +3,21 @@ _.mixin(require("underscore.deep"));
...
+// Encode errors using their properties instead of toJSON()
+function replaceErrors(key, value) {
+  if (value instanceof Error) {
+    return value.toString();
+  }
+
+  return value;
+}
+
 // Converts a map to a string space-delimited key=val pairs
```
Comment (mohit) [inline lib/kayvee.ts]: derp. This comment is `old`

**Comment by mohit** [comment]
LGTM. nit about an old comment not changed as part of this pR.

**Pull request merged**